### PR TITLE
xds: use Cluster_AUTO DnsLookupFamily

### PR DESCRIPTION
**Pull request by zirain**
fixes: https://github.com/envoyproxy/gateway/issues/4605

### CODE REVIEW

**Comment by zirain** [comment]
test passed on https://github.com/envoyproxy/gateway/pull/4690

File: internal/xds/translator/cluster.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func buildEndpointType(settings []*ir.DestinationSetting) EndpointType {
 }
 
 func buildXdsCluster(args *xdsClusterArgs) *clusterv3.Cluster {
-	dnsLookupFamily := clusterv3.Cluster_V4_ONLY
+	dnsLookupFamily := clusterv3.Cluster_AUTO
 	if args.ipFamily != nil && *args.ipFamily == egv1a1.DualStack {
```
Comment (arkodg) [inline internal/xds/translator/cluster.go:88]: should L88 be extended to ?
```
if args.ipFamily != nil {
  switch egv1a1.DualStack:
  .....
 }
 ```

Comment (zirain) [reply internal/xds/translator/cluster.go:88]: sounds good

File: internal/xds/translator/cluster.go [hunk 2]
```diff
@@ -84,9 +84,16 @@ func buildEndpointType(settings []*ir.DestinationSetting) EndpointType {
...
```
Comment (arkodg) [inline internal/xds/translator/cluster.go:88]:
```suggestion
	if args.ipFamily != nil {
```

**Review by arkodg — APPROVED** [review]
thanks !

**Review by zhaohuabing — APPROVED** [review]

**Pull request merged**